### PR TITLE
3905 facilities list

### LINKED ISSUE #3905: GAPS: Add a page to central server which lists all facilities
Part of epic #3660 

## Acceptance criteria
* [ ] On central server only - a list of all stores and facilities in the name table (`type`)
* [ ] The list shows the name of the facility (from the name table) 
* [ ] The list shows the code of the facility
* [ ] A graphQL endpoint is available (only when viewed from central server) which lists all stores

## Describe the solution you'd like 🎁
Create a graphQL query to list all names
ListView for names

- Name, Code, Type, Booleans for isStore, isDonor, isCustomer, isPurchaser, 

New Nav Item Under New `Admin` section (`Admin` to be renamed to Settings?)

### Moneyworks Jobcode 🧰
`CCEIDEV`

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #3905

# 👩🏻‍💻 What does this PR do?

Adds facilities list view when logged in as central server:

Screenshot 2024-05-24 at 12 01 56 PM

## 💌 Any notes for the reviewer?

- Went with `Manage` rather than `Admin` for now, so as not to have to make changes to existing `Admin` piece for now
- Turns out there was no permissions restriction on which names you can see, so just reusing existing names query
  - Is it worth raising an issue for the fact that anyone can query patient data?
- Existing pattern is to show the home icon if the name is a store... should I stick with that or have an isStore column?

# 🧪 Testing

- Log into central server
- [ ] There is a new `Manage` section in the Nav, with a `Facilities` page
- [ ] Facilities page lists all stores and facilities
- [ ] You can search by code
- [ ] You can search by name
- Log into remote site
- [ ] There is no `Manage` nav section

# 📃 Documentation

- [x] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [x] **These areas should be updated or checked**: 
  1. Screenshots of `Settings` in the app bar/Administration pages - rename to Settings
  2.

### CODE REVIEW

**Review by jmbrunskill — APPROVED** [review]
Looks great.
Probably should have a discussion about the icon and location for the `Manage` section but I this is good where it is for now.

**Comment by jmbrunskill** [comment]
> * Went with `Manage` rather than `Admin` for now, so as not to have to make changes to existing `Admin` piece for now

Looks fine to me, although would be nice to have a different icon and maybe rename the other one settings?

> * Turns out there was no permissions restriction on which names you can see, so just reusing existing names query
>   * Is it worth raising an issue for the fact that anyone can query patient data?

Yes please!

> * Existing pattern is to show the home icon if the name is a store... should I stick with that or have an isStore column?
I like it, seems good to me.
At some point we might want to add a filter for store, donor, etc but I think it's totally fine for this PR.

**Comment by mark-prins** [comment]
oh! I was going to start a discussion to get feedback on adding the management section. it's already here!
two quick comments: all the other menu items are nouns, and 'manage' is a verb and this troubles me.. to be consistent it should be 'Management' and I'm really not sure that this works.
also the icon is the same as the admin/settings and that is potentially confusing - we need an icon that says 'manage' without being a 'manager' 🤔

File: client/packages/system/src/Name/api/hooks/document/useNames.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export const useNames = (type: 'customer' | 'supplier') => {
         first: queryParams.first,
         offset: queryParams.offset,
         sortBy: queryParams.sortBy,
-        type: type === 'customer' ? 'customer' : 'supplier',
+        type: type,
```
Comment (roxy-dao) [inline client/packages/system/src/Name/api/hooks/document/useNames.ts:15]: Nit: Just type~?

File: client/packages/common/src/intl/locales/en/manage.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "placeholder.enter-an-item-code-or-name": "Enter item code or name"
```
Comment (mark-prins) [inline client/packages/common/src/intl/locales/en/manage.json:2]: now that this is in two namespaces, putting in `common.json` would be better

File: client/packages/host/src/routers/ManageRouter.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  .addPart(AppRoute.Facilities)
+  .addWildCard()
+  .build();
+
+export const ManageRouter: FC = () => {
+  const gotoCustomers = useMatch(fullFacilitiesPath);
+
+  if (gotoCustomers) {
+    return <NameService />;
+  }
```
Comment (mark-prins) [inline client/packages/host/src/routers/ManageRouter.tsx:19]: :nit
```suggestion
export const ManageRouter: FC = () => {
  const gotoFacilities = useMatch(fullFacilitiesPath);

  if (gotoFacilities) {
    return <NameService />;
  }
```

should that be `gotoNames` or `<FacilityService />` 🤔

File: client/packages/common/src/intl/locales/en/manage.json (hunk 2 above)
Comment (mark-prins) [reply client/packages/common/src/intl/locales/en/manage.json:2]: although.. I see now that it is `Enter facility code or name` really.

File: server/repository/src/db_diesel/name.rs [hunk 4]
```diff
@@ -45,6 +47,8 @@ pub struct NameFilter {
     pub address2: Option<StringFilter>,
     pub country: Option<StringFilter>,
     pub email: Option<StringFilter>,
+
+    pub code_or_name: Option<StringFilter>,
```
Comment (mark-prins) [inline server/repository/src/db_diesel/name.rs:51]: I can sense the request coming in to have also the boolean filters for supplier/customer/donor.

**Review by mark-prins — APPROVED** [review]
Great - thanks! the verdict is hazy re naming and even positioning. 
My summary - having a different icon is good. I think the sliders are not intuitive but they are better than nothing. The name can be changed later, and while I don't like 'Manage' in context, it appears to be the preferred. 

Here's [the branch](https://github.com/msupply-foundation/open-msupply/compare/3905-facilities-list...3905-facilities-list(suggestion)) which I had the icon in, in case it helps

Side note: I have some funky stuff in my datafile..

Screenshot 2024-05-27 at 3 22 16 PM

**Comment by lache-melvin** [comment]
> Side note: I have some funky stuff in my datafile..
> 
> Screenshot 2024-05-27 at 3 22 16 PM

Oh... is this just from test data or should there be some more filtering??

**Comment by mark-prins** [comment]
> Oh... is this just from test data or should there be some more filtering??

I was wondering if we needed filtering. it's not deliberate test data - but there are a lot of blanks in the table.. surely we don't need to anticipate that happening?

**Comment by lache-melvin** [comment]
> I was wondering if we needed filtering. it's not deliberate test data - but there are a lot of blanks in the table.. surely we don't need to anticipate that happening?

Probably from integration tests right? Doesn't _seem_ like the kind of thing we should expect to need to handle in the wild? 🤷‍♀️

**Comment by mark-prins** [comment]
could well be integration tests. they did populate my datafile with a lot of crud.

**Comment by lache-melvin** [comment]
FYI leaving the command K settings shortcut as `a` for now despite the rename to settings... we already have a bunch of `s` related shortcuts, so I'm not sure what would be best to change it to, if at all?

**Comment by adamdewey** [comment]
I can see non-Store suppliers in the list which I think shouldn't be there?

image

**Comment by adamdewey** [comment]
Actually, I'll raise a separate issue for it

**Pull request merged**